common/src/main/java/org/conscrypt/ConscryptFileDescriptorSocket.java

```diff
@@ -216,6 +216,9 @@ public void startHandshake() throws IOException {
 
             try {
                 ssl.doHandshake(Platform.getFileDescriptor(socket), getSoTimeout());
+
+                // Update the session from the current state of the SSL object.
+                sslSession.onHandshakeCompleted(getHostnameOrIP(), getPort());
```

Comment:
Why was this moved? Are we guaranteed that the handshake is completed after a single call to `doHandshake`?